src/status_im/subs/wallet/screens/input_amount.cljs

```diff
@@ -0,0 +1,360 @@
+ (fn [suggested-routes]
+   (when suggested-routes
+     (or (:best suggested-routes) []))))
+
+(defn- every-network-value-is-zero?
+  [sender-network-values]
+  (every? (fn [{:keys [total-amount]}]
+            (and
+             total-amount
+             (money/bignumber? total-amount)
+             (money/equal-to total-amount
+                             (money/bignumber "0"))))
+          sender-network-values))
+
+(rf/reg-sub :send-input-amount-screen/unsupported-token-in-receiver?
+ :<- [:wallet/wallet-send-receiver-network-values]
+ :<- [:wallet/wallet-send-tx-type]
+ (fn [[receiver-network-values
+       tx-type]]
+   (and (not= tx-type :tx/bridge)
+        (->> receiver-network-values
+             (remove #(= (:type %) :add))
+             (every? #(= (:type %) :not-available))))))
+
+(rf/reg-sub :send-input-amount-screen/no-routes-found?
+ :<- [:wallet/wallet-send-sender-network-values]
+ :<- [:send-input-amount-screen/routes]
+ :<- [:wallet/wallet-send-loading-suggested-routes?]
+ :<- [:send-input-amount-screen/unsupported-token-in-receiver?]
+ (fn [[sender-network-values
+       routes
+       loading-routes?
+       unsupported-token-in-receiver?]]
+   (and
+    (every-network-value-is-zero? sender-network-values)
+    (some? routes)
+    (not loading-routes?)
+    (not unsupported-token-in-receiver?))))
+
+(rf/reg-sub :send-input-amount-screen/sending-to-unpreferred-networks?
+ :<- [:wallet/wallet-send-receiver-networks]
+ :<- [:wallet/wallet-send-receiver-preferred-networks]
+ (fn [[receiver-networks
+       receiver-preferred-networks]]
+   (some (comp not (set receiver-preferred-networks)) receiver-networks)))
+
+(defn- insufficient-asset-amount?
+  [{:keys [token-symbol owned-eth-token upper-limit-equals-input-value? no-routes-found? limit-exceeded?
+           sender-network-values enough-assets?]}]
+  (let [eth-selected?   (= token-symbol (string/upper-case constants/mainnet-short-name))
+        zero-owned-eth? (money/equal-to (:total-balance owned-eth-token) 0)
+        exceeded-input? (if eth-selected?
+                          upper-limit-equals-input-value?
+                          zero-owned-eth?)]
+    (and (or no-routes-found? limit-exceeded?)
+         (seq sender-network-values)
+         (or exceeded-input? (not enough-assets?)))))
+
+(rf/reg-sub :send-input-amount-screen/not-enough-asset?
+ :<- [:wallet/wallet-send-sender-network-values]
+ :<- [:wallet/wallet-send-enough-assets?]
+ :<- [:wallet/wallet-send-token]
+ :<- [:send-input-amount-screen/upper-limit-equals-input-value?]
+ :<- [:send-input-amount-screen/no-routes-found?]
+ :<- [:send-input-amount-screen/upper-limit-exceeded?]
+ :<- [:send-input-amount-screen/enabled-from-chain-ids]
+ (fn [[sender-network-values
+       enough-assets?
+       {token-symbol :symbol}
+       upper-limit-equals-input-value?
+       no-routes-found?
+       limit-exceeded?
+       enabled-from-chain-ids]]
+   (let [owned-eth-token (rf/sub [:wallet/token-by-symbol
```

Comment:
`rf/sub` is stateful because it deref's an reaction, but sub handlers should be pure.